src/resource/font.c

```diff
@@ -161,16 +162,36 @@ static void try_create_mutex(SDL_mutex **mtx) {
 	}
 }
 
+static void *ft_alloc(FT_Memory mem, long size) {
+	return mem_alloc(size);
+}
+
+static void ft_free(FT_Memory mem, void *block) {
+	mem_free(block);
+}
+
+static void *ft_realloc(FT_Memory mem, long cur_size, long new_size, void *block) {
+	return mem_realloc(block, new_size);
+}
+
 static void init_fonts(void) {
 	FT_Error err;
 
+	static typeof(*(FT_Memory)0) ft_mem = {
```

Comment:
Freetype is weird. First of all `FT_Memory` is typedef'd to `struct FT_MemoryRec_*`, so it's a pointer type. [The documentation](https://freetype.org/freetype2/docs/reference/ft2-system_interface.html#ft_memoryrec) implies there should be a `typedef struct FT_MemoryRec_ FT_MemoryRec`, but for some reason they decided to declare that struct without a typedef. I could use `struct FT_MemoryRec_` directly, but I guess I was a bit hesitant because that just *looks* like an "implementation detail; not for public use" kind of name. Realistically they're probably never changing that without breaking a lot of code, so I don't mind replacing the ugly `typeof()` here if you prefer.